### ISSUE TITLE
Can't Download rhcos 4.5.6 files

### ISSUE THREAD

**Issue by kazuhisya** [kazuhisya/openshift-ansible-kvm#27]
```
TASK [download coreos kernel] ***************************************************************************************************
Tuesday 25 August 2020  14:26:21 +0900 (0:00:00.051)       0:00:03.033 ******** 
FAILED - RETRYING: download coreos kernel (3 retries left).
FAILED - RETRYING: download coreos kernel (2 retries left).
FAILED - RETRYING: download coreos kernel (1 retries left).
fatal: [127.0.0.1]: FAILED! => {"attempts": 3, "changed": false, "msg": "Unable to find a checksum for file 'rhcos-4.5.6-x86_64-installer-kernel-x86_64' in 'https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.5/4.5.6/sha256sum.txt'"}
```

**Comment by kazuhisya** [comment]
Make `get_url` fail the Hush comparison because https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.5/4.5.6/sha256sum.txt  has the same hash value with different filenames.
This problem has never existed before.

For example in 4.5.2 the hash value is unique.
https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.5/4.5.2/sha256sum.txt

**Comment by kazuhisya** [comment]
`checksum_map` is overwritten with the file name corresponding to the checksum written after the file, so the file name written first does not seem to be compared...
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/get_url.py#L505-L521

**Comment by kazuhisya** [comment]
get_url is unable to find a checksum for file · Issue #71420 · ansible/ansible
https://github.com/ansible/ansible/issues/71420

**Comment by kazuhisya** [comment]
Interim workaround: 

```diff
diff --git a/vars/download.yml b/vars/download.yml
index 9a708a9..f21c63d 100644
--- a/vars/download.yml
+++ b/vars/download.yml
@@ -28,18 +28,21 @@ config:
   coreos_kernel: >-
     {%- if   openshift.dist == 'okd' -%}                                   fedora-coreos-{{ openshift.coreos_version }}-live-kernel-x86_64
     {%- elif openshift.coreos_version | regex_search("^4\.2\.") -%}        rhcos-{{ openshift.coreos_version }}-x86_64-installer-kernel
+    {%- elif openshift.coreos_version | regex_search("^4\.5\.6") -%}       rhcos-installer-kernel-x86_64
     {%- elif openshift.coreos_version | regex_search("^4\.+([3-5])\.") -%} rhcos-{{ openshift.coreos_version }}-x86_64-installer-kernel-x86_64
     {%- else -%}                                                           rhcos-{{ openshift.coreos_version }}-x86_64-installer-kernel-x86_64
     {%- endif -%}
   coreos_initramfs: >-
     {%- if   openshift.dist == 'okd' -%}                                   fedora-coreos-{{ openshift.coreos_version }}-live-initramfs.x86_64.img
     {%- elif openshift.coreos_version | regex_search("^4\.2\.") -%}        rhcos-{{ openshift.coreos_version }}-x86_64-installer-initramfs.img
+    {%- elif openshift.coreos_version | regex_search("^4\.5\.6") -%}       rhcos-installer-initramfs.x86_64.img
     {%- elif openshift.coreos_version | regex_search("^4\.+([4-5])\.") -%} rhcos-{{ openshift.coreos_version }}-x86_64-installer-initramfs.x86_64.img
     {%- else -%}                                                           rhcos-{{ openshift.coreos_version }}-x86_64-installer-initramfs.x86_64.img
     {%- endif -%}
   coreos_bios: >-
     {%- if   openshift.dist == 'okd' -%}                                   fedora-coreos-{{ openshift.coreos_version }}-metal.x86_64.raw.xz
     {%- elif openshift.coreos_version | regex_search("^4\.2\.") -%}        rhcos-{{ openshift.coreos_version }}-x86_64-metal-bios.raw.gz
+    {%- elif openshift.coreos_version | regex_search("^4\.5\.6") -%}       rhcos-metal.x86_64.raw.gz
     {%- elif openshift.coreos_version | regex_search("^4\.+([3-5])\.") -%} rhcos-{{ openshift.coreos_version }}-x86_64-metal.x86_64.raw.gz
     {%- else -%}                                                           rhcos-{{ openshift.coreos_version }}-x86_64-metal.x86_64.raw.gz
     {%- endif -%}
```

**Comment by kazuhisya** [comment]
This issue was solved with upstream ansible.
We can wait for the backport.
https://github.com/ansible/ansible/commit/159544610e5a925d589198febaa5198778ea51c0

**Comment by kazuhisya** [comment]
workaround: 
This is a workaround until the fix(ansible/ansible@1595446) is incorporated into Fedora / RHEL.

```
$ git checkout fix-get_url
$ ./fix-handle_same_sha_for_dl_files.sh
```
